datanode/sqlstore/teams.go

```diff
@@ -256,6 +256,18 @@ func (t *Teams) RefereeJoinedTeam(ctx context.Context, referee *entities.TeamMem
 func (t *Teams) RefereeSwitchedTeam(ctx context.Context, referee *entities.RefereeTeamSwitch) error {
 	defer metrics.StartSQLQuery("Teams", "RefereeSwitchedTeam")()
 
+	// in case the party was removed from the team owner from a team
+	if len(referee.ToTeamID) <= 0 {
+		_, err := t.Exec(
+			ctx,
+			"DELETE FROM team_members WHERE party_id = $1",
```

Comment:
I think if we go and delete all the entries for that party it'll cause issues with the `ListRefereeHistory()` API.

Reply:
Right. Let me check that API.